pkg/search/search.go

```diff
@@ -0,0 +1,137 @@
+package search
+
+import (
+	"fmt"
+)
+
+type Searcher interface {
+	Definition(word string) (string, error)
+	Definitions(words []string) (string, error)
+}
+
+type Sources struct {
+	LocalDictionary
+	OnlineDictionary
+	WikipediaSummary
+	SliderValues []float64
+}
+
+type LocalDictionary struct {
+	// sliderValue float64
+}
+type OnlineDictionary struct {
+	// sliderValue float64
+}
+type WikipediaSummary struct {
+	// sliderValue float64
+}
```

Comment:
There's a pretty elegant approach pattern in Go you could use for this. Let me know if you want me to give you more info, but a hint is that it relies on `Context`.

Reply:
I’ve passed the searchOptions through Context with the new commits in this PR, but I’m not sure if you were referring to that when you said that there is a way to solve it in a more elegant manner.
I usually avoid passing values through context since:
- it’s not immediately apparent from the method signature that it needs an object that is passed through ctx
- to be honest, I’ve heard different opinions - even at work, someone said that if the value that you pass it’s not a constant then you shouldn’t pass it through a ctx. Not quite sure I agree on that one
- another way to solve passing options around would be to use the option pattern maybe? https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md

Maybe I completely missed the elegant approach anyways :laughing:

Reply:
Check out https://medium.com/@matryer/context-keys-in-go-5312346a868d

You're doing almost the exact same thing, but there's a few helpers you can add to make the code nicer.